Modules/Common/AptosCoinLikeModule.php

```diff
@@ -0,0 +1,190 @@
+    public ?CurrencyFormat $currency_format = CurrencyFormat::AlphaNumeric;
+    public ?CurrencyType $currency_type = CurrencyType::FT;
+    public ?FeeRenderModel $fee_render_model = FeeRenderModel::None;
+    public ?array $special_addresses = ['the-contract'];
+    public ?PrivacyModel $privacy_model = PrivacyModel::Transparent;
+    public ?bool $ignore_sum_of_all_effects = false;
+
+    public ?array $events_table_fields = ['block', 'transaction', 'sort_key', 'time', 'currency', 'address', 'effect', 'failed', 'extra'];
+    public ?array $events_table_nullable_fields = ['extra'];
+
+    public ?ExtraDataModel $extra_data_model = ExtraDataModel::Default;
+
+    public ?array $currencies_table_fields = ['id', 'name', 'symbol', 'decimals'];
+    public ?array $currencies_table_nullable_fields = [];
+
+    public ?bool $should_return_events = true;
+    public ?bool $should_return_currencies = true;
+    public ?bool $allow_empty_return_events = true;
+    public ?bool $allow_empty_return_currencies = true;
+
+    public ?bool $mempool_implemented = false;
+    public ?bool $forking_implemented = false;
+
+    final public function pre_initialize()
+    {
+        $this->version = 1;
+    }
+
+    final public function post_post_initialize()
+    {
+
+    }
+
+    final public function pre_process_block($block_id)
+    {
+        $block = requester_single($this->select_node(), endpoint: "v1/blocks/by_height/{$block_id}?with_transactions=true", timeout: $this->timeout);
+        $this->block_time = date('Y-m-d H:i:s', (int) ((int) $block['block_timestamp'] / 1000000));
+
+        $events = [];
+        $currencies_to_process = [];
+        $sort_key = 0;
+
+        foreach ($block['transactions'] as $trx)
+        {
+            if ($trx['type'] !== 'user_transaction')
+            {
+                continue;
+            }
+
+            $failed = false;
+            if ($trx['vm_status'] !== 'Executed successfully')
+            {
+                $failed = true;
+            }
+
+            $diff_sum = [];
+            foreach ($trx['changes'] as $change)
+            {
+                $changed_resource = $change['data']['type'] ?? '';
+                if (!str_starts_with($changed_resource, '0x1::coin::CoinStore<'))
+                {
+                    continue;
+                }
+                $coin = str_replace(['0x1::coin::CoinStore<', '>'], '', $changed_resource);
+                if ($coin === '0x1::aptos_coin::AptosCoin')
+                {
+                    continue;
+                }
+
+                $address = $change['address'];
+                $changed_resource_encode = urlencode($changed_resource);
+                try
+                {
+                    $resource_before = requester_single($this->select_node(), endpoint: "v1/accounts/{$address}/resource/{$changed_resource_encode}?ledger_version={$block['first_version']}", timeout: $this->timeout);
+                } catch (RequesterException $e)
```

Comment:
This is not a good practice. What if the API just goes down? The same exception will be thrown.

Reply:
Fixed with `valid_codes` and parsing the curl result.